File: 프로그래머스/unrated/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.py
def solution(r1, r2):
    answer = 1
    for r in range(1, r2):
        y1 = -(-((r1 ** 2 - r ** 2) ** 0.5) // 1) if r1 > r else 0
        y2 = ((r2 ** 2 - r ** 2) ** 0.5) // 1
        answer += y2 - y1 + 1
    return answer * 4

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.js
function solution(r1, r2) {
    let answer = 0;
    for (let i = 1; i <= r2; i++) {
        if (i <= r1) {
            const r1_Y = (r1 * r1 - i * i) ** 0.5;
            const r2_Y = (r2 * r2 - i * i) ** 0.5;
            answer += Math.floor(r2_Y) - Math.ceil(r1_Y) + 1;
        } else {
            const r2_Y = (r2 * r2 - i * i) ** 0.5;
            answer += Math.floor(r2_Y) + 1;
        };
    };
    
    return answer * 4;
}